internal/provider/kubernetes/config/default/config_patch.yaml

```diff
@@ -2,18 +2,18 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: envoy-gateway
-  namespace: system
+  namespace: envoy-gateway-system
```

Comment:
This change is unneeded due to https://github.com/envoyproxy/gateway/blob/main/internal/provider/kubernetes/config/default/kustomization.yaml#L1-L2

Reply:
uncommenting the above line changes the name of the `envoy-gateway` pod to `envoy-gateway-envoy-gateway`, do we want that ?